src/bvals/comms/boundary_communication.cpp

```diff
@@ -138,6 +138,15 @@ TaskStatus SendBoundBufs(std::shared_ptr<MeshData<Real>> &md) {
           sending_nonzero_flags(b) = non_zero[0] || non_zero[1] || non_zero[2];
         });
       });
+  // 1. Parallel scan per rank to get the starting indices of the buffers
+
+  // 2. Check the size of the buffer (how do you do this without extra DtoH call?) and
+  // possibly allocate more storage
+  //    [Alternatively could just allocate to maximal size initially]
+
+  // 3. Pack the combined buffers
+
+  // 4. Send the combined buffers
```

Comment:
Is this what's already been done or what's now been implemented?

Does "maximal size initially" refer to sparsity? So for dense code it's always "maximal size"

Since the 

1. and 2. can be done at AMR rebuilding and before going through the ghosts if we use always allocate "maximal size". For Dense codes this is always the case. We could assign the indices determined by the parallel scan to `bndinfo.buf`. 

3. Is only relevant for sparse codes? I know we can do this in-parallel and in-place easily for one-team per MPI Rank. Packing with more than one team per rank might be trickier.

Reply:
Yes, by "maximal size" I mean just allocating the combined buffers during remeshing with enough space to store all of the buffers (even if they are sparse and the field itself is unallocated on a given block), and by "initially" I mean at remeshing. As you say, points 1 and 2 are only necessary if you want to dynamically allocate space for sparse field buffers (which I think is something to worry about after the hackathon).

Reply:
resolving, as this is outdated.